pkg/kf/commands/root.go

```diff
@@ -38,14 +38,26 @@ func NewKfCommand() *cobra.Command {
 		Long: templates.LongDesc(`
       kf is like cf for Knative
       `),
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			// Read configuration from either config or home directory
+			if err := p.ReadConfig(); err != nil {
+				return err
+			}
+
+			p.ApplyDefaults(config.KfParams{
```

Comment:
In lieu of my other comment: it feels like this logic should be in the config package